app/src/components/ScriptText.js

```diff
@@ -34,7 +36,14 @@ const ScriptText = React.memo(
                     <LocalIdentityBadge badgeOnly={disabled} compact entity={part} />
                   </span>
                 ) : (
-                  <span key={index}>{part}</span>
+                  transformIPFSHash(part, (word, isIpfs, i) => {
+                    const ipfsUrl = generateURI(word)
+                    return isIpfs ? (
+                      <Link href={ipfsUrl}>{ipfsUrl} </Link>
+                    ) : (
+                      <span key={index}>{word}</span>
+                    )
+                  })
```

Comment:
What do you think of reafctoring this to something like: 
```suggestion
                    if (isIpfs) {
                      const ipfsUrl = generateURI(word)
                       return <Link href={ipfsUrl}>{ipfsUrl} </Link>
                    }
              
                     return <span key={index}>{word}</span>
                  })
```

So we don't generate the URI in case it's not an ipfsHash